Library/Homebrew/extend/os/mac/diagnostic.rb

```diff
@@ -90,6 +91,18 @@ def check_clt_up_to_date
         EOS
       end
 
+      def check_xcode_8_without_clt_on_el_cap
+        return unless MacOS::Xcode.without_clt?
+        # Scope this to Xcode 8 on El Cap for now
+        return unless MacOS.version == :el_capitan && MacOS::Xcode.version >= "8"
+
+        <<-EOS.undent
+          You have Xcode 8 installed without the CLT; this causes certain builds to fail.
```

Comment:
Agreed it's probably best to explicitly mention El Cap. It'll only be a matter of time until someone sticks the message on stackoverflow or something without much context 🙈.